core/core_integrations/slack.js

```diff
@@ -17,596 +15,553 @@ inTransaction = {},
-				numSocketsToShutDown--;
-				return;
-			}
-			if (!shuttingDown) {
-				init(slackTeams[connectionDetails.team_id].token, connect);
-			}
-		}).on('error', function(data) {
-			console.debug("slack-> received error:\n" + data);
-		});
-	}
-	else {
-		exports.start(eventReceivedCallback);
-	}
+startPingPongTimer = function(teamId) {
+    var team = teamData[teamId],
+        messageId = ++team.ping_id;
+
+    if (!messageId) {
+        messageId = team.ping_id = 1;
+    }
+
+    team.timeSincelastMessageRecieved = getTime();
+    team.pingTimeout = setTimeout(function() {
+        if (getTime() - team.timeSincelastMessageRecieved >= 60000 && messageId === team.ping_id && !shuttingDown) {
+            console.debug('slack-> sending ping');
+            sendPing(teamId);
+        }
+    }, 60000);
 },
 
-openPrivateMessage = function(message, thread, senderId) {
-	var slackTeams = exports.config.slack_teams,
-		teamInfo = getChannelIdAndTeamId(thread);
-
-	if (teamInfo.token !== null) {
-		var body = {
-			"token": teamInfo.token,
-			"user": senderId
-		};
-
-		request({
-			"uri": 'https://slack.com/api/im.open',
-			"method": 'GET',
-			"qs": body
-		},
-		function (error, response, body) {
-			body = JSON.parse(body);
-			if (response.statusCode !== 200) {
-				console.debug('slack-> error: ' + response.statusCode);
-			}
-			else if (!body.ok) {
-				console.debug('slack-> Failed to send message to user with id: ' + teamInfo.channel_id + ', error: ' + body.error);
-			}
-			else {
-				slackTeams[teamInfo.team_id].channels[body.channel] = {"channel": body.channel, "user": senderId};
-				sendMessage(message, createThreadId(body.channel.id, teamInfo.team_id));
-			}
-		});
-	}
-	else {
-		console.debug('slack-> No slack team found!!!');
-	}
+pong = function(teamId) {
+    console.debug('slack-> recieved pong');
+    teamData[teamId].timeSincelastMessageRecieved = getTime();
+    teamData[teamId].pong_id++;
 },
 
-eventReceived = function(event, teamId) {
-	if (shuttingDown) return;
-	switch (event.type) {
-		case 'message':
-			recMessage(event, teamId);
-			break;
-		case 'channel_created':
-			channelCreated(event, teamId);
-			break;
-		case 'channel_deleted':
-			channelDeleted(event, teamId);
-			break;
-		case 'channel_rename':
-			channelRename(event, teamId);
-			break;
-		case 'im_created':
-			channelCreated(event, teamId);
-			break;
-		case 'group_joined':
-			channelCreated(event, teamId);
-			break;
-		case 'group_close':
-			channelDeleted(event, teamId);
-			break;
-		case 'group_rename':
-			channelRename(event, teamId);
-			break;
-		case 'user_change':
-			userChange(event, teamId);
-			break;
-		case 'team_join':
-			userChange(event, teamId);
-			break;
-		case 'team_rename':
-			teamRename(event, teamId);
-			break;
-		case 'pong':
-			pong(event, teamId);
-			break;
-		default:
-			console.debug("slack-> Message of type " + event.type + " not supported");
-			break;
-	}
+findSlackBot = function(body, slackTeam) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['findSlackBot' is defined but never used.](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707054/issues/source?bid=3282987&fileBranchId=3288593#l302)